submission/proxy_server.go

```diff
@@ -35,9 +36,9 @@ type ProxyServer struct {
 }
 
 // NewProxyServer creates ProxyServer instance. Call Run() to init.
-func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout time.Duration) *ProxyServer {
+func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout time.Duration, mf monitoring.MetricFactory) *ProxyServer {
 	s := &ProxyServer{addTimeout: reqTimeout}
-	s.p = NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder)
+	s.p = NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder, mf)
```

Comment:
not in this diff (which is fine), but did you think about just writing:

```go
return &ProxyServer{
  addTimeout: reqTimeout,
  p: NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder, mf)
}
```

Reply:
When you say 'not in this diff' does it mean you'd prefer a separate PR resolving it?

Reply:
No, I just meant I was commenting on something you didn't introduce in this PR. No need to make a separate PR if you want to fix it now.

Reply:
And, no need to fix if you think it's just fine as it is.